frontend/src/metabase/static-viz/components/CategoricalWaterfallChart/CategoricalWaterfallChart.jsx

```diff
@@ -74,13 +68,14 @@ const CategoricalWaterfallChart = ({
   accessors = POSITIONAL_ACCESSORS,
   settings,
   labels,
+  getColor,
 }) => {
   const entries = calculateWaterfallEntries(
     data,
     accessors,
     settings?.showTotal,
   );
-  const colors = settings?.colors;
+  const colors = settings?.colors ?? {};
```

Comment:
Does this change imply `settings?.colors` could be `null`? Since `colors` will be passed to

https://github.com/metabase/metabase/blob/5e9f03ae4b8bd40cae09ab375b5715d3b5b9ce82/frontend/src/metabase/static-viz/lib/colors.ts#L18-L26

as the first argument which already has a default value set. The only reason `?? {}` would be needed is when passing `null` which should crash `getWaterfallColors`.

If that's the case I think it's better to not let people assume things and type `settings.colors` as nullable.